src/algebra/order/to_interval_mod.lean

```diff
@@ -181,69 +181,115 @@ by { rw [to_Ioc_mod_eq_iff hp, set.right_mem_Ioc], exact ⟨lt_add_of_pos_right
 by rw [add_comm, to_Ioc_div_add_zsmul, add_comm]
 
+/-! Note we omit `to_Ioc_div_zsmul_add'` as `-m + to_Ioc_div hp a b` is not very convenient. -/
+
 @[simp] lemma to_Ico_div_sub_zsmul (a b : α) (m : ℤ) :
   to_Ico_div hp a (b - m • p) = to_Ico_div hp a b - m :=
 by rw [sub_eq_add_neg, ←neg_smul, to_Ico_div_add_zsmul, sub_eq_add_neg]
 
+@[simp] lemma to_Ico_div_sub_zsmul' (a b : α) (m : ℤ) :
+  to_Ico_div hp (a - m • p) b = to_Ico_div hp a b + m :=
+by rw [sub_eq_add_neg, ←neg_smul, to_Ico_div_add_zsmul', sub_neg_eq_add]
+
 @[simp] lemma to_Ioc_div_sub_zsmul (a b : α) (m : ℤ) :
   to_Ioc_div hp a (b - m • p) = to_Ioc_div hp a b - m :=
 by rw [sub_eq_add_neg, ←neg_smul, to_Ioc_div_add_zsmul, sub_eq_add_neg]
 
+@[simp] lemma to_Ioc_div_sub_zsmul' (a b : α) (m : ℤ) :
+  to_Ioc_div hp (a - m • p) b = to_Ioc_div hp a b + m :=
+by rw [sub_eq_add_neg, ←neg_smul, to_Ioc_div_add_zsmul', sub_neg_eq_add]
+
 @[simp] lemma to_Ico_div_add_right (a b : α) : to_Ico_div hp a (b + p) = to_Ico_div hp a b + 1 :=
 by simpa only [one_zsmul] using to_Ico_div_add_zsmul hp a b 1
 
+@[simp] lemma to_Ico_div_add_right' (a b : α) : to_Ico_div hp (a + p) b = to_Ico_div hp a b - 1 :=
+by simpa only [one_zsmul] using to_Ico_div_add_zsmul' hp a b 1
+
 @[simp] lemma to_Ioc_div_add_right (a b : α) : to_Ioc_div hp a (b + p) = to_Ioc_div hp a b + 1 :=
 by simpa only [one_zsmul] using to_Ioc_div_add_zsmul hp a b 1
 
+@[simp] lemma to_Ioc_div_add_right' (a b : α) : to_Ioc_div hp (a + p) b = to_Ioc_div hp a b - 1 :=
+by simpa only [one_zsmul] using to_Ioc_div_add_zsmul' hp a b 1
+
 @[simp] lemma to_Ico_div_add_left (a b : α) : to_Ico_div hp a (p + b) = to_Ico_div hp a b + 1 :=
 by rw [add_comm, to_Ico_div_add_right]
 
+@[simp] lemma to_Ico_div_add_left' (a b : α) : to_Ico_div hp (p + a) b = to_Ico_div hp a b - 1 :=
+by rw [add_comm, to_Ico_div_add_right']
+
 @[simp] lemma to_Ioc_div_add_left (a b : α) : to_Ioc_div hp a (p + b) = to_Ioc_div hp a b + 1 :=
 by rw [add_comm, to_Ioc_div_add_right]
 
+@[simp] lemma to_Ioc_div_add_left' (a b : α) : to_Ioc_div hp (p + a) b = to_Ioc_div hp a b - 1 :=
+by rw [add_comm, to_Ioc_div_add_right']
+
 @[simp] lemma to_Ico_div_sub (a b : α) : to_Ico_div hp a (b - p) = to_Ico_div hp a b - 1 :=
 by simpa only [one_zsmul] using to_Ico_div_sub_zsmul hp a b 1
 
+@[simp] lemma to_Ico_div_sub' (a b : α) : to_Ico_div hp (a - p) b = to_Ico_div hp a b + 1 :=
+by simpa only [one_zsmul] using to_Ico_div_sub_zsmul' hp a b 1
+
 @[simp] lemma to_Ioc_div_sub (a b : α) : to_Ioc_div hp a (b - p) = to_Ioc_div hp a b - 1 :=
 by simpa only [one_zsmul] using to_Ioc_div_sub_zsmul hp a b 1
 
-lemma to_Ico_div_sub' (a b c : α) : to_Ico_div hp a (b - c) = to_Ico_div hp (a + c) b :=
+@[simp] lemma to_Ioc_div_sub' (a b : α) : to_Ioc_div hp (a - p) b = to_Ioc_div hp a b + 1 :=
+by simpa only [one_zsmul] using to_Ioc_div_sub_zsmul' hp a b 1
+
+lemma to_Ico_div_sub_eq (a b c : α) : to_Ico_div hp a (b - c) = to_Ico_div hp (a + c) b :=
```

Comment:
I'm not super happy with this new name (previously `to_Ico_div_sub'`). Some other options are:
* `to_Ico_div_sub''` (if in doubt, add more primes)
* `to_Ico_div_sub_eq_to_Ico_div_add` (perhaps too long?)

Reply:
I think `to_Ico_div_sub_eq_to_Ico_div_add` because it's not so clear which should be the LHS and which should be the RHS, so having both `to_Ico_div_sub` and `to_Ico_div_add` with the name will with the search. But yeah I agree it's annoyingly long (but gets better with CamelCase).

Reply:
Done, this scheme seems to work nicely for the other clashes too